File: Interpreter/Golang/AndExpression.go
package main

import "strings"

type AndExpression struct {
	expression1 IExpression
	expression2 IExpression
}

func (andExpression *AndExpression) solve(context string) bool {
	if !strings.Contains(context, "&&") {
		return false
	}

	return andExpression.expression1.solve(context) && andExpression.expression2.solve(context)
} 

File: Interpreter/Dart/AndExpression.dart
import 'IExpression.dart';

class AndExpression implements IExpression {
  IExpression? _expression1, _expression2;

  AndExpression(this._expression1, this._expression2);

  @override
  bool solve(String context) {
    if (!context.contains("&&")) {
      return false;
    }

    return _expression1!.solve(context) && _expression2!.solve(context);
  }
}
